### PR TITLE
fix: verification tweaks

### PR DESCRIPTION
**Pull request by MacDeluca**
# Summary of Changes

This pull request introduces a more robust and explicit account verification system by replacing checks based on the presence of a refresh token with a new `VerificationStatus` enum and associated helper functions. The changes improve the accuracy and reliability of user verification logic throughout the codebase, add comprehensive tests, and ensure that edge cases (such as cancelled or expired credentials) are handled correctly.

It also implements a small fix to ensure the biometrics prompt is only registered once.

# Acceptance Criteria

Replace this text with the acceptance criteria that must be met for this PR to be approved.

# Screenshots, videos, or gifs

Replace this text with embedded media for UI changes if they are included in this PR. If there are none, simply enter N/A

# Related Issues

Replace this text with tagged issue #'s that are relevant to this PR. If there are none, simply enter N/A

### CODE REVIEW

File: app/src/bcsc-theme/hooks/useSecureActions.tsx [hunk 1]
```diff
@@ -655,6 +656,21 @@ export const useSecureActions = () => {
...
+   */
+  const isVerified = useCallback(
+    (credential: CredentialInfo | null, refreshToken?: string): boolean => {
+      if (!credential && refreshToken) {
+        // Potential bug detected: Missing credential but refresh token exists? Log a warning for visibility, but treat as not verified to avoid false positives.
+        logger.warn('[HydrateVerifiedState] No credential found but refresh token exists — treating as not verified.')
+      }
+
+      // If credential exists and event is not Cancel or Expire -> verified
+      return !!credential && credential.bcscEvent !== 'Cancel' && credential.bcscEvent !== 'Expire'
```
Comment (bryce-mcmath) [inline app/src/bcsc-theme/hooks/useSecureActions.tsx]: We will have to retest app expiry and agent cancellation

**Review by jleach — APPROVED** [review]
Tested. Verification works as expected.
